src/compiler/checker.ts

```diff
@@ -18420,6 +18444,46 @@ namespace ts {
             return checkPropertyAccessExpressionOrQualifiedName(node, node.left, node.right);
         }
 
+        function getPropertyByPrivateName(apparentType: Type, leftType: Type, right: PrivateName): Symbol | undefined {
+            let classWithShadowedPrivateName;
+            let klass = getContainingClass(right);

```

Comment:
```suggestion
            let container = getContainingClass(right);
```
(along with other references)